docs/faq.rst

```diff
@@ -788,17 +788,40 @@ functions, for example :func:`jax.nn.softmax` can replace uses of
 :func:`jax.numpy.sign`, :func:`jax.nn.softplus` can replace uses of
 :func:`jax.nn.relu`, etc.
 
+How do I convert a JAX Tracer to a NumPy Array?
+-----------------------------------------------
+When inspecting a transformed JAX function at runtime, you'll find that array
+values are replaced by :class:`jax.core.Tracer` objects::
 
-Additional Sections
--------------------
+  @jax.jit
+  def f(x):
+    print(type(x))
+    return x
+
+  f(jnp.arange(5))
+  # <class 'jax.interpreters.partial_eval.DynamicJaxprTracer'>
+
+A frequent question is how such a tracer can be converted back to a normal NumPy
+array. In short, **it is impossible to convert a Tracer to a NumPy array**, because
+a tracer is an abstract representation of *every possible* value with a given shape
+and dtype, while a numpy array is a concrete member of that abstract class.
+For more discussion of this, see `How to think in JAX`_.
+
+This question usually comes up in the context of another goal, related to
+accessing intermediate values in a computation at runtime. For example:
 
-.. comment We refer to the anchor below in JAX error messages
+- If you wish to print a traced value at runtime for debugging purposes, you might
+  consider using :func:`jax.debug.print`.
+- If you wish to call non-JAX code within a transformed JAX function, you might
+  consider using :func:`jax.pure_callback`, an example of which is available at
+  `Pure callback example`_.
 
-``Abstract tracer value encountered where concrete value is expected`` error
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-See :class:`jax.errors.ConcretizationTypeError`
```

Comment:
Note: deleted this because the error message that references it was removed many JAX versions ago.